core/src/main/scala/org/apache/spark/scheduler/TaskResult.scala

```diff
@@ -77,14 +78,12 @@ private[spark] class DirectTaskResult[T](
    *
    * After the first time, `value()` is trivial and just returns the deserialized `valueObject`.
    */
-  def value(): T = {
+  def value(resultSer: SerializerInstance = null): T = {
     if (valueObjectDeserialized) {
       valueObject
     } else {
-      // This should not run when holding a lock because it may cost dozens of seconds for a large
-      // value.
-      val resultSer = SparkEnv.get.serializer.newInstance()
-      valueObject = resultSer.deserialize(valueBytes)
+      valueObject = Option(resultSer).getOrElse(SparkEnv.get.serializer.newInstance())
```

Comment:
nit: I prefer the following codes because `Option(...).getOrElse` doesn't improve the readability but creates unnecessary objects.
```Scala
val resultSer = if (resultSer == null) SparkEnv.get.serializer.newInstance() else resultSer	
valueObject = resultSer.deserialize(valueBytes)
```